src/array.ts

```diff
@@ -230,29 +230,51 @@ export const select = <T, K>(
   }, [] as K[])
 }
 
+type GetterFunc<T extends number | object> = (item: T) => number
+
 /**
  * Max gets the greatest value from a list
  *
- * Ex. max([{ num: 1 }, { num: 2 }], x => x.num) == 2
+ * @example
+ * max([ 2, 3, 5]) == 5
+ * max([{ num: 1 }, { num: 2 }], x => x.num) == 2
  */
-export const max = <T extends number | object>(
+export function max<T extends number>(
   array: readonly T[],
-  getter?: (item: T) => number
-) => {
-  const get = getter ? getter : (v: any) => v
+  getter?: undefined
+): T | null
+export function max<T extends object>(
```

Comment:
```suggestion
export function max<T>(
```
This lets us take in any type that is _not_ `number` in this overload, so we can handle arrays, and string, and other things.